index.js

```diff
@@ -103,7 +104,7 @@ module.exports = function enforceHTTPS(options) {
 
     // build redirect url
     var httpsHost = options.hostname || url.parse('http://' + this.request.header.host).hostname;
-    var redirectTo = 'https://' + httpsHost + ':' + options.port;
+    var redirectTo = 'https://' + httpsHost + (options.skipPort ? '' : ':' + options.port);
```

Comment:
PS. I know this will require much more changes in tests but this is best solution I can think of. My goal is to make this middleware as simple, safe and easy to use as possible. I also think it will be nice to have this feature on by default.